velox/expression/CastExpr.cpp

```diff
@@ -689,12 +689,23 @@ void CastExpr::evalSpecialForm(
   auto fromType = inputs_[0]->type();
   auto toType = std::const_pointer_cast<const Type>(type_);
 
+  SelectivityVector remainingRows = rows;
+  context.deselectErrors(remainingRows);
+  // Skip evaluation if all requested rows have errors. Return a NULL vector.
+  if (!remainingRows.hasSelections()) {
+    context.ensureWritable(remainingRows, toType, result);
```

Comment:
nit: `if (!result) { result = BaseVector::createNullConstant(toType, rows.end(), context.pool()); }` for optimization.